XenAdmin/Wizards/ConversionWizard/VmSelectionPage.cs

```diff
@@ -139,17 +166,72 @@ private void BulkCheck(bool check)
             try
             {
                 _updating = true;
-
                 foreach (SourceVmRow row in dataGridViewVms.Rows)
+                {
                     row.IsChecked = check;
+                }
             }
             finally
             {
                 _updating = false;
+                SupportedVersionCheck();
                 UpdateButtons();
             }
         }
 
+        private void SupportedVersionCheck()
+        {
+            tableLayoutPanelError.Visible = false;
+            foreach (SourceVmRow row in dataGridViewVms.Rows)
+            {
+                if (row.IsChecked && !row.IsSupportedGuest())
+                {
+                    pictureBoxError.Image = Images.StaticImages._000_Alert2_h32bit_16;
+                    labelError.Text = Messages.CONVERSION_UNSUPPORTED_VM_SELECTED_WARNING;
+                    tableLayoutPanelError.Visible = true;
+                    return;
+                }
+            }
+        }
+
+        private void UpdateComponentEnabledStatusWhenRefresh(bool enabled)
+        {
+            buttonSelectAll.Enabled = enabled;
+            buttonClearAll.Enabled = enabled;
+            buttonRefresh.Enabled = enabled;
+            showOnlySupportedGuestCheckBox.Enabled = enabled;
+        }
+
+        private bool IsSupportedGuest(VmInstance vm)
+        {
+            if (vm.SupportedXSVersions == null)
+            {
+                return true;
+            }
+            foreach (string supportedVersion in vm.SupportedXSVersions)

```

Comment:
This has confused me a bit: `SupportedXSVersions` are the XenServer versions where the guest is supported, right? Then, should we not compare it to the version of the _Pool_ where the VPX is running instead of the VPX version?

Reply:
It should be compared XenServer version instead of the VPX version. Have already fixed.